controller/dataplane/controller_reconciler_utils.go

```diff
@@ -78,6 +84,169 @@ func (r *Reconciler) ensureDataPlaneAddressesStatus(
 	return false, nil
 }
 
+// ensureKonPluginInstallation ensures that the KongPluginInstallation resources
+// referenced by the DataPlane are resolved and DataPlane is configured to use them.
+// During resolving for each DataPlane based on each instance of KongPluginInstallation
+// ConfigMap is created and mounted. The DataPlane manages its lifecycle.
+func ensureKongPluginInstallation(
```

Comment:
Can we change the name of this `func` (and the godoc on it) to mention that it returns a `Deployment` generation opt rather than an actual function that would ensure the KPI for a `DataPlane`? I believe this might be misleading to future readers.

Altenatively we could split the opt code from code that does prepares the `ConfigMap`s and deletes the old ones.

WDYT?

Reply:
I reworked it like that, please tell WDYT